engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/BasedOnVersionAvroSchemaDeterminer.scala

```diff
@@ -0,0 +1,35 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry
+
+import cats.data.Validated
+import cats.data.Validated.Valid
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.avro.AvroSchemaDeterminer
+
+class BasedOnVersionAvroSchemaDeterminer(createSchemaRegistryClient: () => SchemaRegistryClient,
+                                         topic: String,
+                                         version: Option[Int]) extends AvroSchemaDeterminer {
+
+  @transient private lazy val schemaRegistryClient: SchemaRegistryClient = createSchemaRegistryClient()
+
+
+  override def determineSchemaInRuntime: Validated[SchemaRegistryError, Option[Schema]] =
+    determineSchemaUsedInTyping.map(Some(_))
+
+  override def determineSchemaUsedInTyping: Validated[SchemaRegistryError, Schema] =
+    schemaRegistryClient
+      .getFreshSchema(topic, version, isKey = false)
+
+}
+
+/**
+ * This implementation uses concrete schema version during typing, but in runtime uses schema from record.
+ * It can be useful when you want to have basic typing before deploy but during runtime don't want to do schema evolution -
+ * just trying to use record as it is.
```

Comment:
So this is the case when:
- I have #input of type schemaA 
- I want to to typed filtering on #input.field1
- I want to pass #input to sink, with all (possibly unknown at deployment time) fields?